File: AtCoder/cd_981A.cxx
// CodeForce Problem 981A

#include <bits/stdc++.h>
using namespace std;

class Solution{
	public:
	static bool is_palindrome(string s){
		string rev = s;
		reverse(rev.begin(),rev.end());
		return rev==s;
	}
	static int longest_nonpal(string &s){
		vector<int> res;
		for(auto i = 0; i < s.size(); i++){
			for(auto j = 1; j <= s.size()-i; j++){
				if(!is_palindrome(s.substr(i,j)))
					res.emplace_back(s.substr(i,j).size());
			}
		}
		return(res.size()!=0 ? *max_element(res.begin(),res.end()) : 0);
	}	
};

int main(){
	ios_base::sync_with_stdio(0);
	cin.tie(0); cout.tie(0);
	string s; cin >> s;
	cout << Solution::longest_nonpal(s);
	return 0;
}

File: AtCoder/cd_981A.py
# CodeForce Problem 981A

def is_palindrome(s:str)->bool:
    rev = str()
    rev = reversed(s)
    return list(rev)==list(s)

def longest_nonpal(s:str)->int:
    res = []
    for i in range(len(s)):
        for j in range(1,len(s)+1-i):
            if(not is_palindrome(s[i:j])):
                res.append(len(s[i:j]))

    if(len(res)==0):
        return 0
    return max(res)

                           

if __name__=='__main__':
    s = input()
    print(longest_nonpal(s))
